lib/spack/spack/environment.py

```diff
@@ -1697,9 +1697,8 @@ def _to_lockfile_dict(self):
             for s in spec.traverse():
                 dag_hash_all = s.build_hash()
                 if dag_hash_all not in concrete_specs:
-                    spec_dict = s.to_node_dict(hash=ht.build_hash)
-                    # Assumes no legacy formats, since this was just created.
-                    spec_dict[ht.dag_hash.name] = s.dag_hash()
+                    spec_dict = s.node_dict_with_hashes(
+                        hash=ht.full_hash, dep_hash=ht.build_hash)
```

Comment:
why not full hash everywhere?

Reply:
Yes, maybe we should.  But then do we have to change the key used in the spack.lock?  And does that mean a new version of the file format we have to support in perpetuity (like when the format changed from dag hash to build hash)?